services/dynamodb/src/test/java/software/amazon/awssdk/services/dynamodb/model/ItemBuilderTest.java

```diff
@@ -0,0 +1,113 @@
+                       .attrs("attrs", Arrays.asList(15, 200L, 15.5, "stringValue", BYTES1, ByteBuffer.wrap(BYTES2), true))
+                       .build();
+
+        assertThat(item, hasEntry("String", AttributeValue.builder().s("stringValue").build()));
+        assertThat(item, hasEntry("int", AttributeValue.builder().n("15").build()));
+        assertThat(item, hasEntry("long", AttributeValue.builder().n("200").build()));
+        assertThat(item, hasEntry("float", AttributeValue.builder().n("15.5").build()));
+        assertThat(item, hasEntry("Integer", AttributeValue.builder().n("15").build()));
+        assertThat(item, hasEntry("Long", AttributeValue.builder().n("200").build()));
+        assertThat(item, hasEntry("Float", AttributeValue.builder().n("15.5").build()));
+        assertThat(item, hasEntry("bool", AttributeValue.builder().bool(true).build()));
+        assertThat(item, hasEntry("Boolean", AttributeValue.builder().bool(true).build()));
+        assertThat(item, hasEntry("StringsVarArgs", AttributeValue.builder().ss("value1", "value2").build()));
+        assertThat(item, hasEntry("StringsCollection", AttributeValue.builder().ss("value1", "value2").build()));
+        assertThat(item, hasEntry("NumbersVarArgs", AttributeValue.builder().ns("1", "2.0", "3").build()));
+        assertThat(item, hasEntry("NumbersCollection", AttributeValue.builder().ns("1", "2.0", "3").build()));
+
+        assertThat(item, hasEntry(equalTo("bytes"), bytesMatching(BYTES1)));
+        assertThat(item, hasEntry(equalTo("ByteBuffer"), bytesMatching(BYTES1)));
+        assertThat(item, hasEntry(equalTo("bytesVarArgs"), bytesMatching(BYTES1, BYTES2)));
+        assertThat(item, hasEntry(equalTo("bytesCollection"), bytesMatching(BYTES1, BYTES2)));
+        assertThat(item, hasEntry(equalTo("ByteBuffersVarArgs"), bytesMatching(BYTES1, BYTES2)));
+        assertThat(item, hasEntry(equalTo("ByteBuffersCollection"), bytesMatching(BYTES1, BYTES2)));
+
+        assertThat(item.get("attrs").l(), hasSize(7));
+        assertThat(item.get("attrs").l().get(0), equalTo(AttributeValue.builder().n("15").build()));
+        assertThat(item.get("attrs").l().get(1), equalTo(AttributeValue.builder().n("200").build()));
+        assertThat(item.get("attrs").l().get(2), equalTo(AttributeValue.builder().n("15.5").build()));
+        assertThat(item.get("attrs").l().get(3), equalTo(AttributeValue.builder().s("stringValue").build()));
+        assertThat(item.get("attrs").l().get(4), bytesMatching(BYTES1));
+        assertThat(item.get("attrs").l().get(5), bytesMatching(BYTES2));
+        assertThat(item.get("attrs").l().get(6), equalTo(AttributeValue.builder().bool(true).build()));
+    }
+
+    private static Matcher<AttributeValue> bytesMatching(byte[] bytes) {
+        return simpleMatcher(item -> byteBufferEquals(item.b(), bytes), String.valueOf(bytes));
```

Comment:
AssertJ has this built in with .as(...). Is there a reason we're using hamcrest here?

Reply:
`AssertJ` doesn't have an `hasEntry` type matcher that doesn't require me to create an actual `Map.Entry` object - unless I missed something....

Reply:
Would `assertThat(item).containsEntry("bytes", myByteArray)` work? 

More examples: https://github.com/joel-costigliola/assertj-examples/blob/master/assertions-examples/src/test/java/org/assertj/examples/MapAssertionsExamples.java#L63